src/sys/stat.rs

```diff
@@ -179,3 +182,70 @@ pub fn fchmodat<P: ?Sized + NixPath>(dirfd: RawFd, pathname: &P, mode: Mode, fla
 
     Errno::result(res).map(drop)
 }
+
+#[cfg(target_os = "linux")]
+mod linux {
+    use {Errno, Result, NixPath};
+    use std::os::unix::io::RawFd;
+    use libc;
+    use fcntl::AtFlags;
+    use sys::time::TimeSpec;
+
+    /// A file timestamp.
+    pub enum UtimeSpec {
+        /// File timestamp is set to the current time.
+        Now,
+        /// The corresponding file timestamp is left unchanged.
+        Omit,
+        Time(TimeSpec)
+    }
+
+    fn to_timespec(time: &UtimeSpec) -> libc::timespec {
```

Comment:
Can you add a doccomment for this function? Nix devs should know if they can or should use it elsewhere in this function.